### PR TITLE
authenticate via the instance_metrics_writer token

### PR DESCRIPTION
**Pull request by zhenz**

### CODE REVIEW

File: cc-cloudfuse [hunk 1]
```diff
@@ -7,13 +7,24 @@ fi
...
-# check auth url
-if ! [ -z ${OS_AUTH_URL:+x} ]; then
-    if [[ $OS_AUTH_URL != *?v3 ]]; then
-        OS_AUTH_URL="$OS_AUTH_URL/v3"
-    fi
+OPENSTACK_VENDOR_DATA='http://169.254.169.254/openstack/latest/vendor_data.json'
+OPENSTACK_VENDOR_DATA_2='http://169.254.169.254/openstack/latest/vendor_data2.json'
+
+JSON_VENDOR_DATA=$(curl -s $OPENSTACK_VENDOR_DATA_2)
+if [ "$JSON_VENDOR_DATA" = '{}' ]; then
```
Comment (diurnalist) [inline cc-cloudfuse:14]: I didn't realize `=` could work, but in fact I learned that `==` is actually a Bash-specific thing, and the POSIX way is `=`. So, this is the best way! Cool.

File: cc-cloudfuse [hunk 2]
```diff
@@ -7,13 +7,24 @@ fi
...
+  JSON_VENDOR_DATA=$(curl -s $OPENSTACK_VENDOR_DATA)
 fi
 
+function extract_json_key {
+  RESULT=$(echo "$2" | sed "s/.*$1\": \"//g" | sed 's/".*//g')
+  echo "$RESULT"
+}
+
+OS_TOKEN=$(extract_json_key "instance_metrics_writer_token" "$JSON_VENDOR_DATA")
+OS_PROJECT_ID=$(extract_json_key "project_id" "$JSON_VENDOR_DATA")
```
Comment (diurnalist) [inline cc-cloudfuse:24]: This is one of my pet peeves with Keystone. It doesn't understand the token is already scoped to a project! Lame.

**Review by diurnalist — APPROVED** [review]
Is there any other bug we may have with parsing the vendor data? What if the endpoint returns nothing? Will this fail badly, or will those params just default to empty strings?

**Comment by zhenz** [comment]
> Is there any other bug we may have with parsing the vendor data? What if the endpoint returns nothing? Will this fail badly, or will those params just default to empty strings?

I tested and it will default to empty strings and the error message will be "Failed to authenticate".

**Pull request merged**